### PR TITLE
Feat: Added GetStatus APIs in SDKs.

### PR DESCRIPTION
**Pull request by dakshbeniwal17**
* Added GetJob APIs to check status of Job in GO,Python and JS SDKs.
* Added `sync` (bool) param in createDeployment APIs:
      sync=true (waits for job to be completed , checks Job status every 10 seconds till the job is completed or failed)
      sync= false (return the deployment response immediately).

### CODE REVIEW

**Review by amanjain97 — APPROVED** [review]

File: pingpong-golang-sdk/example/main.go [hunk 1]
```diff
@@ -11,24 +10,27 @@ import (
...
+}
+
+func RunGetJobExample(client *sdk.Client) {
+	jobId := "fd2d1136-3da3-421d-83bf-cf5e9c528302"
+	job, err := client.Deployments().GetJob(jobId)
+	if err != nil {
+		panic(err)
+	}
+
+	log.Println("job: ", job)
```
Comment (EwanValentine) [inline pingpong-golang-sdk/example/main.go:35]: Can we remove this?

File: pingpong-golang-sdk/sdk/deployments/deployments.go [hunk 2]
```diff
@@ -57,6 +57,22 @@ func (c *Client) Create(deployment CreateDeployment) (*Deployment, error) {
 	if err := json.Unmarshal(response, &deploymentResponse); err != nil {
 		return nil, errors.Wrap(err, "failed to unmarshal deployment response")
 	}
+	status := deploymentResponse.Job.Status
+	eta := deploymentResponse.Job.Eta
+	if deployment.Sync {
+		for status != "complete" && status != "failed" && eta > 0 {
```
Comment (EwanValentine) [inline pingpong-golang-sdk/sdk/deployments/deployments.go:63]: Can we make "complete" and "failed" constants?

File: pingpong-python-sdk/pingpongsdk/deployments/deployments.py [hunk 3]
```diff
@@ -27,14 +29,26 @@ def create(self, deployment: CreateDeployment) -> Deployment:
             model = self.models.get_by_id(deployment.model)
             id = model.id
             request = request_dto(id, deployment)
-
+            sync=deployment.sync
             deployment = super().post("/api/v1/deployments", request)
-            return dto(deployment)
+            response=dto(deployment)
+            status=response.status
+            eta= response.job['eta']
```
Comment (EwanValentine) [inline pingpong-python-sdk/pingpongsdk/deployments/deployments.py:36]: nit: spacing is off here, some have spaces, some don't, and eta has one space

File: pingpong-python-sdk/pingpongsdk/deployments/deployments.py [hunk 4]
```diff
@@ -27,14 +29,26 @@ def create(self, deployment: CreateDeployment) -> Deployment:
...
+            if sync: 
+                while status!='complete' and status!='failed' and eta > 0:
```
Comment (EwanValentine) [inline pingpong-python-sdk/pingpongsdk/deployments/deployments.py:38]: Maybe constants here as well

File: pingpong-typescript-sdk/src/lib/deployments/deployments.ts [hunk 5]
```diff
@@ -48,6 +63,20 @@ class Deployments extends Client {
             }
         }
     }
+
+    async get_job(id: string): Promise<Job | undefined> {
```
Comment (EwanValentine) [inline pingpong-typescript-sdk/src/lib/deployments/deployments.ts:67]: JS is typically `getJob` rather than snake case

**Pull request merged**